File: benchmarks/python/brownfcn.py
import numpy as np

def brownfcn(x: np.ndarray) -> np.ndarray:
    """
    Computes the value of the Brown benchmark function.

    Parameters:
    -----------
    x (ndarray): Input array of shape (M, N) where M is the number of points and N is the number of dimensions.

    Returns:
    --------
    ndarray: Array of size (M, 1) containing the function value for each corresponding row of x.
    
    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at
    Google's e-mail service or feel free to kindly modify the repository.
    """
    
    n = x.shape[1]
    scores = np.zeros((x.shape[0], 1))
    x = x ** 2
    
    for i in range(n-1):
        scores += x[:, i] ** (x[:, i+1] + 1) + x[:, i+1] ** (x[:, i] + 1)
    
    return scores

File: _includes/brownfcn.m
% Computes the value of the Brown benchmark function.
% SCORES = BROWNFCN(X) computes the value of the Brown function at point X.
% BROWNFCN accepts a matrix of size M-by-N and returns a vetor SCORES of 
% size M-by-1 in which each row contains the function value for the 
% corresponding row of X.
% 
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = brownfcn(x)
    
    n = size(x, 2);  
    scores = 0;
    
    x = x .^ 2;
    for i = 1:(n-1)
        scores = scores + x(:, i) .^ (x(:, i+1) + 1) + x(:, i+1).^(x(:, i) + 1);
    end
end